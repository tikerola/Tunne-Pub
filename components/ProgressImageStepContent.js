import React from "react";
import { View, Text, StyleSheet, FlatList, Image } from "react-native";
import { colors } from "../constants/colors";
import CustomButton from "./CustomButton";

export default function ProgressImageStepContent({
  navigation,
  navigateTo,
  data,
  handleSubmit,
  pad = false,
}) {
  return (
    <View style={pad && styles.pad}>
      <Text style={styles.instructionText}>
        Hyv√§! Valitse nyt{" "}
        <Text
          style={styles.listButtonText}
          onPress={() => navigation.navigate(navigateTo)}
        >
          tunnekuvista
        </Text>
        <Text style={styles.instructionText}>
          {" "}
          ne kuvat, jotka parhaiten kuvaavat tunteitasi.
        </Text>
      </Text>
      <FlatList
        style={[styles.flatlist, { marginTop: 15, marginBottom: 20 }]}
        horizontal
        data={data}
        keyExtractor={(item, index) => index.toString()}
        renderItem={({ item }) => {
          return (
            <Image testID="image" style={styles.image} source={item.filename} />
          );
        }}
      />
      {data.length > 0 && (
        <CustomButton
          title="Tallenna"
          onPress={() => handleSubmit("dummy text")}
        />
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  listButtonText: {
    fontSize: 20,
    color: "#3a89ed",
    textDecorationLine: "underline",
  },

  flatlist: {
    marginTop: 0,
    marginBottom: 0,
    width: "90%",
    alignSelf: "center",
  },

  flatlistItem: {
    fontSize: 20,
    margin: 5,
    color: "red",
    fontFamily: "Segoe-Print",
  },

  instructionText: {
    fontSize: 18,
    paddingLeft: 15,
    paddingRight: 15,
    fontFamily: "Roboto-Light",
    color: "#bbb",
  },
  pad: {
    backgroundColor: colors.stepperContentBg,
    padding: 50,
    borderRadius: 10,
  },
  image: {
    width: 96,
    height: 140,
    marginRight: 5,
  },
});
