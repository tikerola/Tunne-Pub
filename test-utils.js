import React from "react";
import { NavigationContext } from "@react-navigation/native";
import { render } from "@testing-library/react-native";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import CARD_DATA from "./data/cards";
import NEEDS_DATA from "./data/needs";
import { SENTENCES_DATA } from "./data/sentences";
import WORD_DATA from "./data/words";

const INITIAL_STATE = {
  cards: {
    // cards: CARD_DATA,
    // words: WORD_DATA,
    // needs: NEEDS_DATA,
    // sentences: SENTENCES_DATA,
    activeSummaryItemId: "123",
    summaryOfChosenItems: {
      123: {
        id: "123",
        name: "",
        cards: [],
        feelingsWords: [],
        feelingsYouMissWords: [],
        needs: ["hei"],
        needsBehindAFeeling: [],
        sentences: [],
        questions: {},
        itemSaved: true,
      },
    },
    screenNumber: 1,
    numOfItems: 0,
    initialLoadFinished: false,
  },
};

const mockStore = configureStore([]);
const store = mockStore(INITIAL_STATE);

const navContext = {
  isFocused: () => true,
  // addListener returns an unscubscribe function.
  addListener: jest.fn(() => jest.fn()),
};

const reduxRender = (ui, options, storeData = store) =>
  render(ui, {
    wrapper: ({ children }) => (
      <Provider store={storeData}>
        <NavigationContext.Provider value={navContext}>
          {children}
        </NavigationContext.Provider>
      </Provider>
    ),
    ...options,
  });

// re-export everything for convenience
export * from "@testing-library/react-native";

// override render method
export { reduxRender as render, store };
