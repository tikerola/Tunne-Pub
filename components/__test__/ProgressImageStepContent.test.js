import { fireEvent } from "@testing-library/react-native";
import React from "react";
import CARD_DATA from "../../data/cards";
import { render } from "../../test-utils";
import ProgressImageStepContent from "../ProgressImageStepContent";

describe("<ProgressImageStepContent />", () => {
  const data = [CARD_DATA[0], CARD_DATA[1], CARD_DATA[2]];

  const navigation = {
    navigate: jest.fn(),
  };

  const navigateTo = "somewhere";
  const handleSubmit = jest.fn();

  it("should match snapshot", () => {
    expect(
      render(
        <ProgressImageStepContent
          navigation={navigation}
          data={data}
          navigateTo={navigateTo}
          handleSubmit={handleSubmit}
        />
      )
    ).toMatchSnapshot();
  });

  it("should render images", () => {
    const component = render(
      <ProgressImageStepContent
        navigation={navigation}
        data={data}
        navigateTo={navigateTo}
        handleSubmit={handleSubmit}
      />
    );

    const images = component.queryAllByTestId("image");
    expect(images.length).toBe(3);
  });

  it("should call handleSubmit when button is pushed", () => {
    const component = render(
      <ProgressImageStepContent
        navigation={navigation}
        data={data}
        navigateTo={navigateTo}
        handleSubmit={handleSubmit}
      />
    );

    fireEvent.press(component.getByText("Tallenna"));
    expect(handleSubmit).toHaveBeenCalled();
  });

  it("should navigate to imageScreen when pushed link", () => {
    const component = render(
      <ProgressImageStepContent
        navigation={navigation}
        data={data}
        navigateTo={navigateTo}
        handleSubmit={handleSubmit}
      />
    );

    fireEvent.press(component.getByText("tunnekuvista"));
    expect(navigation.navigate).toHaveBeenCalledWith(navigateTo);
  });
});
