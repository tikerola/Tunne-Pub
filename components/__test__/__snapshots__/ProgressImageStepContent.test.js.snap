// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressImageStepContent /> should match snapshot 1`] = `
<View
  style={false}
>
  <Text
    style={
      Object {
        "color": "#bbb",
        "fontFamily": "Roboto-Light",
        "fontSize": 18,
        "paddingLeft": 15,
        "paddingRight": 15,
      }
    }
  >
    Hyv√§! Valitse nyt
     
    <Text
      onPress={[Function]}
      style={
        Object {
          "color": "#3a89ed",
          "fontSize": 20,
          "textDecorationLine": "underline",
        }
      }
    >
      tunnekuvista
    </Text>
    <Text
      style={
        Object {
          "color": "#bbb",
          "fontFamily": "Roboto-Light",
          "fontSize": 18,
          "paddingLeft": 15,
          "paddingRight": 15,
        }
      }
    >
       
      ne kuvat, jotka parhaiten kuvaavat tunteitasi.
    </Text>
  </Text>
  <RCTScrollView
    data={
      Array [
        Card {
          "feeling": "happy",
          "filename": 1,
          "id": "1",
        },
        Card {
          "feeling": "sad",
          "filename": 1,
          "id": "2",
        },
        Card {
          "feeling": "worried",
          "filename": 1,
          "id": "3",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {
          "alignSelf": "center",
          "marginBottom": 0,
          "marginTop": 0,
          "width": "90%",
        },
        Object {
          "marginBottom": 20,
          "marginTop": 15,
        },
      ]
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 140,
              "marginRight": 5,
              "width": 96,
            }
          }
          testID="image"
        />
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 140,
              "marginRight": 5,
              "width": 96,
            }
          }
          testID="image"
        />
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 140,
              "marginRight": 5,
              "width": 96,
            }
          }
          testID="image"
        />
      </View>
    </View>
  </RCTScrollView>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#3a89ed",
        "borderRadius": 5,
        "elevation": 5,
        "margin": 0,
        "opacity": 1,
        "padding": 5,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.44,
        "shadowRadius": 5,
        "width": 100,
      }
    }
    testID="button"
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "white",
          "fontFamily": "Roboto-Light",
          "fontSize": 16,
        }
      }
      testID="title"
    >
      Tallenna
    </Text>
  </View>
</View>
`;
